src/libsystemd-network/dhcp6-option.c

```diff
@@ -529,6 +535,14 @@ int dhcp6_option_parse_ia(sd_dhcp6_client *client, DHCP6Option *iaoption, DHCP6I
                 if (len < DHCP6_OPTION_IA_NA_LEN)
                         return -ENOBUFS;
 
+                /* According to RFC8415, IAs which do not match the client's IAID should be ignored,
+                 * but not necessary to ignore or refuse the whole message. */
+                if (((const struct ia_na*) iaoption->data)->id != iaid)
+                        /* ENOANO indicates the option should be ignored. */
+                        return log_dhcp6_client_errno(client, SYNTHETIC_ERRNO(ENOANO),
+                                                      "Received IA_NA option whose IAID is different "
+                                                      "from the one choosed by the client, ignoring.");
```

Comment:
"Received an IA_NA option with a different IAID "

/choosed/chosen/

Reply:
Ouch. Fixed. Thank you!